processing/src/main/java/io/druid/segment/Rowboat.java

```diff
@@ -126,6 +126,11 @@ public int compareTo(Rowboat rhs)
     return retVal;
   }
 
+  public int compareTimestamp(Rowboat rhs)
```

Comment:
used in IndexMerger.merge when determining rowMergerFn, if rollup=false,only order by timestamp
```
        if (rollup) {
          return CombiningIterable.create(
              new MergeIterable<Rowboat>(
                  Ordering.<Rowboat>natural().nullsFirst(),
                  boats
              ),
              Ordering.<Rowboat>natural().nullsFirst(),
              new RowboatMergeFunction(sortedMetricAggs)
          );
        } else {
          return new MergeIterable<Rowboat>(
              new Ordering<Rowboat>()
              {
                @Override
                public int compare(Rowboat left, Rowboat right)
                {
                  return left.compareTimestamp(right);
                }
              }.nullsFirst(),
              boats
          );
        }
```